src/lib/Microsoft.Health.Common/Storage/BlobContainerClientFactory.cs

```diff
@@ -28,19 +28,15 @@ public BlobContainerClient CreateStorageClient(BlobContainerClientOptions option
             {
                 return new BlobContainerClient(containerUri.ToString(), blobUri.BlobContainerName);
             }
+            else if (provider != null)
+            {
+                var tokenCredential = provider.GetCredential();
+                return new BlobContainerClient(containerUri, tokenCredential);
+            }
             else
             {
                 throw new Exception($"Unable to create blob container client for {blobUri}");
```

Comment:
I think we should be more explicit as to why this failed so it's easier to debug.

Reply:
Yep makes sense. Updated with error messages that explain the invalid config / how we got to this point.